### PR TITLE
chore: use next eslint and add a workflow to lint and format on push

### PR DESCRIPTION
**Pull request by smakosh**
## What's new?
- Use Next ESlint and use their recommended approach to handle formatting using prettier
> See: https://nextjs.org/docs/basic-features/eslint
- Switch to absolute imports
- Add a github action workflow to lint and format on push

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on: [push]
+jobs:
+  build:
+    name: lint and format
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node: ['14.x']
+        os: [ubuntu-latest, windows-latest, macOS-latest]
```
Comment (tarrencev) [inline .github/workflows/lint.yml:11]: hmm is there a reason to run on all these os's?

**Review by tarrencev — APPROVED** [review]
minor comment otherwise lgtm!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (smakosh) [reply .github/workflows/lint.yml:11]: Nope, this is usually done when building libraries. Let me stick with one os.

**Review by subimage — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**